src/UglyToad.PdfPig.Fonts/CompactFontFormat/CharStrings/Type2CharStrings.cs

```diff
@@ -11,7 +11,7 @@
     /// as the local (per font) and global (per font set) subroutines.
     /// The CharStrings are lazily evaluated.
     /// </summary>
-    internal class Type2CharStrings
+    internal sealed class Type2CharStrings
```

Comment:
Any particular reason to seal these? Does it provide performance benefits?

Reply:
It can offer performance improvements, see here https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-6/#peanut-butter

As a general rule of thumb, I always seal my classes, with the possibility to unseal it if need be. My reasoning is "_I need a reason for a class to not be sealed_", instead of "_I need a reason for a class to be sealed_".

But let me know what you think, happy to discuss

Reply:
Thanks, in this case it's fine since these are internal classes. I think for public facing classes it would be a breaking API change to start sealing them if people are relying on creating derived types which we wouldn't be able to detect. I'll flag any public instances.